### ISSUE TITLE
Environment can only contain strings

### ISSUE THREAD

**Issue by jakirkham** [conda/conda-build#1853]
It appears we are getting some errors in some builds where `conda-build` is unable to create the build environment on Windows Python 2.7 builds. No issues on other platforms or other Python versions. Have included the traceback from one such build below and a link to the AppVeyor log. This is using `conda-build` 2.1.6, which I'm aware is not latest. However, we are blocked from releasing 2.1.7 by a potentially related issue ( https://github.com/conda-forge/conda-forge-build-setup-feedstock/issues/60 ).

```python
Traceback (most recent call last):
  File "C:\Miniconda\Scripts\conda-build-script.py", line 5, in <module>
    sys.exit(conda_build.cli.main_build.main())
  File "C:\Miniconda\lib\site-packages\conda_build\cli\main_build.py", line 322, in main
    execute(sys.argv[1:])
  File "C:\Miniconda\lib\site-packages\conda_build\cli\main_build.py", line 313, in execute
    noverify=args.no_verify)
  File "C:\Miniconda\lib\site-packages\conda_build\api.py", line 97, in build
    need_source_download=need_source_download, config=config)
  File "C:\Miniconda\lib\site-packages\conda_build\build.py", line 1494, in build_tree
    config=config)
  File "C:\Miniconda\lib\site-packages\conda_build\build.py", line 1013, in build
    create_env(config.build_prefix, specs, config=config)
  File "C:\Miniconda\lib\site-packages\conda_build\build.py", line 689, in create_env
    execute_actions(actions, index, verbose=config.debug)
  File "C:\Miniconda\lib\site-packages\conda\plan.py", line 643, in execute_actions
    inst.execute_instructions(plan, index, verbose)
  File "C:\Miniconda\lib\site-packages\conda\instructions.py", line 135, in execute_instructions
    cmd(state, arg)
  File "C:\Miniconda\lib\site-packages\conda\instructions.py", line 78, in LINK_CMD
    link(state['prefix'], dist, lt, index=state['index'])
  File "C:\Miniconda\lib\site-packages\conda\install.py", line 1009, in link
    if not run_script(prefix, dist, 'post-link'):
  File "C:\Miniconda\lib\site-packages\conda\install.py", line 500, in run_script
    subprocess.check_call(args, env=env)
  File "C:\Miniconda\lib\subprocess.py", line 181, in check_call
    retcode = call(*popenargs, **kwargs)
  File "C:\Miniconda\lib\subprocess.py", line 168, in call
    return Popen(*popenargs, **kwargs).wait()
  File "C:\Miniconda\lib\subprocess.py", line 390, in __init__
    errread, errwrite)
  File "C:\Miniconda\lib\subprocess.py", line 640, in _execute_child
    startupinfo)
TypeError: environment can only contain strings
```

ref: https://ci.appveyor.com/project/conda-forge/pyqt-feedstock/build/1.0.41/job/32polar37fo6smoq

cc @ocefpaf @gillins

**Comment by SylvainCorlay** [comment]
Also, the new ` conda-build 2.1.7`  appears to be broken too.

I have a `build.sh` script that only does `cp -r $SRC_DIR/include/ $PREFIX/include/`. Files are put in the right place with conda-build 2.1.4 or 2.1.6 and not 2.1.7 where it lands into ` $PREFIX/include/include`...

@jakirkham would you guys consider pinning conda-forge do `2.1.4`?

**Comment by msarahan** [comment]
It does seem like pinning to 2.1.4 may be the lesser of all evils.  Sorry
for the breakage.  Please improve our test suite if you have time.

**Comment by jakirkham** [comment]
We can pin `conda-build` without removing the other packages, but yes we should do this short term. I'll put up a PR and let you guys know.

**Comment by SylvainCorlay** [comment]
I know you guys are stretching really thin and I really appreciate all the work even though I feel the pain of the breakage.

**Comment by jakirkham** [comment]
This should do the trick ( https://github.com/conda-forge/conda-forge-build-setup-feedstock/pull/61 ).

**Comment by jakirkham** [comment]
AFAICT pinning `conda-build` to 2.1.4 seems to have effectively worked around this issue. It also seems to have worked around a similar issue with the upload script at conda-forge. Perhaps they share a common cause.

xref: https://github.com/conda-forge/conda-forge-build-setup-feedstock/issues/60

**Comment by jschueller** [comment]
I suspect https://github.com/conda-forge/matplotlib-feedstock/issues/114 to re a related bug ; it disappeared as well.

**Comment by msarahan** [comment]
fixed by #1861 - closing.
